primap2/tests/test_data_reading.py

```diff
@@ -46,6 +46,82 @@ def test_metadata_for_variable(unit, entity, expected_attrs):
     )
 
 
+@pytest.mark.parametrize(
+    "to_test_for_float, expected_result",
+    [
+        ("0.15", True),
+        ("IE", False),
+        ("NaN", True),
+        (
+            "25,000",
+            False,
+        ),  # be careful when reading data to process thousands seperators
+        ("25,00", False),
+        ("IE, NO", False),
+    ],
+)
+def test_is_float(to_test_for_float, expected_result):
+    assert pm2io._data_reading.is_float(to_test_for_float) == expected_result
```

Comment:
This is test is nice, just as inspiration: it is also possible to do doctests like I did in the docstring of `preferred_unit`. They are also automatically run etc. Sometimes easier to do since you can define them directly in the docstring of the function instead of making a special test function in another file. For simple input -> output tests doctests work great, for more complex tests a separate function is definitely better.